### PR TITLE
Add ENV `ENABLE_IMAP`

### PR DESCRIPTION
**Pull request by casperklein**
# Description

This PR adds the ENV `ENABLE_IMAP`. Like `ENABLE_POP3`, this allows to enable/disable IMAP.

In a POP3 only setup, this allows to not start the IMAP service.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] **I have added information about changes made in this PR to `CHANGELOG.md`**

### CODE REVIEW

**Comment by casperklein** [comment]
This test failed: https://github.com/docker-mailserver/docker-mailserver/actions/runs/7235052163/job/19712079567

Re-running the test was successful.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,12 @@ All notable changes to this project will be documented in this file. The format
 
 ### Added
 
+- **Dovecot:**
+  - ENV `SASL_PASSWD` ([#2946](https://github.com/docker-mailserver/docker-mailserver/pull/2946))
```
Comment (polarathene) [inline CHANGELOG.md:12]: Placeholder?

**Review by polarathene — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (casperklein) [reply CHANGELOG.md:12]: Oups, you are right.

**Review by polarathene — APPROVED** [review]

**Comment by georglauterbach** [comment]
Are `ENABLE_POP3` and `ENABLE_IMAP` both zero or both one valid configurations?

**Comment by casperklein** [comment]
Yes. Like most other ENVs, the values are only checked if equal 1 to enable it. All other values are considered to disable it. I haven't changed the logic, only streamlined the docs.

**Review by georglauterbach — APPROVED** [review]
👍🏼

**Pull request merged**